src/sentry/auth/helper.py

```diff
@@ -379,6 +383,25 @@ def _post_login_redirect(self) -> HttpResponseRedirect:
 
         return response
 
+    def handle_authentication_using_verification_key(self, identity, verification_key):
+        cluster = redis.clusters.get("default").get_local_client_for_key(_REDIS_KEY)
+        verification_value_byte = cluster.hgetall(verification_key)
+        if verification_value_byte:
+            verification_value = {
+                y.decode("ascii"): verification_value_byte.get(y).decode("ascii")
+                for y in verification_value_byte.keys()
+            }
+            if verification_value["identity_id"] == identity["id"]:
+                op = "verify"
+                user = User.objects.get(id=verification_value["user_id"])
```

Comment:
instead of using the user here, why not just use the user that's on the session and just simply check that they are the same? same with member below.

Reply:
like above, theres no user defined in the session so i had to retrieve it